### PR TITLE
DOP-3355: fix LG component styling

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-3355

### Current Behavior:

cloud-docs
https://www.mongodb.com/docs/atlas/

docs-landing
https://www.mongodb.com/docs/

guides
https://www.mongodb.com/docs/guides/

### Staging Links:

atlas (cloud-docs)
https://docs-mongodbcom-integration.corp.mongodb.com/cdb7e3f32a7928d2ed5f5fbc62245710a908266d/master/cloud-docs/seung.park/DOP-3355/

landing (docs-landing)
https://docs-mongodbcom-integration.corp.mongodb.com/cdb7e3f32a7928d2ed5f5fbc62245710a908266d/master/landing/seung.park/DOP-3355/

guides (docs-tutorials)
https://docs-mongodbcom-integration.corp.mongodb.com/cdb7e3f32a7928d2ed5f5fbc62245710a908266d/master/guides/seung.park/DOP-3355/

### Notes:
some styling we applied was getting overwritten by LG styling so want to verify these are intended: 
- sidenav left caret - should it be middle aligned or top aligned for multi line TOC items? this opens up the question for the "as" property - when components are rendered "as" something else, which styling should take precedence?
image

- headings were the focal point of this PR with [margins not being applied ](https://www.mongodb.com/docs/atlas/setup-cluster-security/), but more cleanups were deemed necessary after explore dependencies of Links

### CODE REVIEW

**Review by allisonmui9 — APPROVED** [review]
Heading styling LGTM. Answering your question in the notes, the caret should be top aligned (what's live today).

**Review by rayangler — CHANGES_REQUESTED** [review]
Pretty much LGTM! Will formally approve once the extra console log is removed

> when components are rendered "as" something else, which styling should take precedence?

In general, I think this would be case by case, depending on the components being used. For something like the side nav items, we'd probably want `SideNavItem`'s styling to take precedence over the `Link` component's styling, since the main goal of the `as` prop in this case is using `Link`'s functionality

File: src/components/Sidenav/GuidesTOCTree.js [hunk 1]
```diff
@@ -12,6 +13,7 @@ const GuidesTOCTree = ({ chapters, guides, handleClick, slug }) => {
   const currentChapterName = guides?.[slug]?.['chapter_name'];
   const guidesInChapter = chapters?.[currentChapterName]?.guides;
   const { activeHeadingId, headingNodes } = useContext(ContentsContext);
+  console.log('GuidesTOCTree');
```
[the remark this answers is not in the document]
Comment (casthewiz) [reply src/components/Sidenav/GuidesTOCTree.js]: I just pushed a commit to remove this 👀

**Review by rayangler — APPROVED** [review]

File: src/components/Link.js [hunk 2]
```diff
@@ -92,7 +97,7 @@ const Link = ({
   const hideExternalIcon = hideExternalIconProp ?? shouldHideExternalIcon;
 
   return (
-    <LGLink className={cx(LGlinkStyling)} href={to} hideExternalIcon={hideExternalIcon} {...other}>
+    <LGLink className={cx([LGlinkStyling])} href={to} hideExternalIcon={hideExternalIcon} {...other}>
```
[the remark this answers is not in the document]
Comment (casthewiz) [reply src/components/Link.js:100]: I want to assume that any iterable args passed in would be spread, but 🤔

**Pull request merged**